fuel/src/main/kotlin/com/github/kittinunf/fuel/core/MultipartBody.kt

```diff
@@ -0,0 +1,165 @@
+package com.github.kittinunf.fuel.core
+
+import com.github.kittinunf.fuel.core.requests.MultipartRequest
+import java.io.ByteArrayInputStream
+import java.io.ByteArrayOutputStream
+import java.io.OutputStream
+import java.net.URLConnection
+import java.nio.charset.Charset
+
+data class MultipartBody(private val request: MultipartRequest) : Body {
+    private var inputAvailable: Boolean = true
+
+    override fun isConsumed() = !inputAvailable
```

Comment:
could this be a val with a private set?

Reply:
`isConsumed` always reflects `inputAvailable`. If it's a val, it will never change, right?

Reply:
sorry I meant 
``` kotlin
var isConsumed :Boolean 
    private set
```